### DISCUSSION TITLE
I made some advanced drag and drop prototype

### DISCUSSION THREAD

**Discussion by enomado** [emilk/egui#1530] [Show and tell]
https://user-images.githubusercontent.com/707007/164800548-8b5a0dc8-8ade-43fc-bd94-6508215637b8.mp4

What do you think?

https://gist.github.com/enomado/07c9024d0e863754e27106ee914e59db

The code is very dirty for now. I just experimenting with egui and suddenly wanted to make this prototype before migration.

After few days with egui I want to say that its awesome.

May be you have advice how to improve. I know I could use data() instead of "property drilling".
Btw I didnt get the idea of ui.scope

**Comment by crumblingstatue** [comment]
I love it! Better drag and drop support would be amazing in egui!

**Reply by Titaniumtown** [reply]
for real! would love something like this natively added to egui

**Comment by enomado** [comment]
Splitted into one file. added load/store

https://gist.github.com/enomado/5e1de10edfa55bf005d95acd0930af2b

**Comment by mjehrhart** [comment]
I must be missing something but what is the difference between this drag and drop and the normal egui drag and drop?

**Reply by crumblingstatue** [reply]
in the normal demo, you can only drag items between columns. You can't control where within the column it will be placed. Here you can reorder items within a column by dragging.

**Reply by mjehrhart** [reply]
Oh okay, thanks for the info.

**Reply by enomado** [reply]
Furthermore, this splits row into normal row, which you can interact, , and drag handle

**Comment by lucasmerlin** [comment]
I made a crate for drag and drop sorting, inspired by your prototype! For now it only supports sorting in a single list, no dragging between lists, but it makes implementing drag and drop a lot more straightforward. 
https://github.com/lucasmerlin/egui_dnd
https://crates.io/crates/egui_dnd
